### ISSUE TITLE
colexecwindow: BenchmarkWindowFunctions/NTILE panics if coldata-batch-size is 568

### ISSUE THREAD

**Issue by stevendanna** [cockroachdb/cockroach#69686]
**Describe the problem**

```
BenchmarkWindowFunctions/NTILE/rows=2272/partition=true/order=true-16                panic: argument of ntile() must be greater than zero

goroutine 165 [running]:
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.ExpectedError(...)
  github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:200
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecwindow.(*nTileWithPartition).seekNextPartition(0xc000635db0, 0x43dde20, 0xc0000eab40, 0x5, 0x1, 0x5)
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecwindow/ntile.eg.go:197 +0x39a
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecwindow.(*bufferedWindowOp).Next(0xc0001a6b00, 0x238, 0xc0000eaba0)
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecwindow/buffered_window.go:279 +0xcb
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecwindow.BenchmarkWindowFunctions.func3.1.1.1.1(0xc00073b440)
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecwindow/window_functions_test.go:1296 +0xe3
testing.(*B).runN(0xc00073b440, 0x1)
  GOROOT/src/testing/benchmark.go:192 +0xeb
testing.(*B).run1.func1(0xc00073b440)
  GOROOT/src/testing/benchmark.go:232 +0x57
created by testing.(*B).run1
  GOROOT/src/testing/benchmark.go:225 +0x7f
I210831 15:10:35.452082 1 (gostd) testmain.go:78  [-] 1  Test //pkg/sql/colexec/colexecwindow:colexecwindow_test exited with error code 2
```

This was a metamorphic build under bazel with `coldata-batch-size` set to 568.  

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_Bench/3388291?showLog=3388291_31663_5088.5099.31573.31663&logFilter=debug

**To Reproduce**

Manually set coldata-batch-size to 568 and run the benchmark.

```
> git diff pkg/col/coldata/batch.go
diff --git a/pkg/col/coldata/batch.go b/pkg/col/coldata/batch.go
index 62c65f7799..9c7278133b 100644
--- a/pkg/col/coldata/batch.go
+++ b/pkg/col/coldata/batch.go
@@ -83,7 +83,8 @@ var _ Batch = &MemBatch{}
 // better, so we decided to keep 1024 as it is a power of 2).
 var defaultBatchSize = int64(util.ConstantWithMetamorphicTestRange(
        "coldata-batch-size",
-       1024, /* defaultValue */
+       568,
+       // 1024, /* defaultValue */
        // min is set to 3 to match colexec's minBatchSize setting.
        3, /* min */
        MaxBatchSize,
```

`make bench 'BENCHES=BenchmarkWindowFunctions/NTILE/rows=\d+/partition=true/order=true' PKG=./pkg/sql/colexec/colexecwindow GOTESTFLAGS="-v"`

Jira issue: CRDB-9722

**Comment by DrewKimball** [comment]
I think this one is already fixed - it was opened only a day or two after #69587 was merged, which makes me think the test build that lead to this error was on a previous version of master (the log has since been deleted so I don't know for sure).
@yuzefovich does this sound right to you?

**Comment by yuzefovich** [comment]
Yeah, seems very likely.

**Comment by DrewKimball** [comment]
Closing as fixed by #69587.
